### PR TITLE
refactor: add devname into gpt.Partition, refactor probe package

### PR DESCRIPTION
**Pull request by Unix4ever**
Necessary refactoring to prepare for disk encryption implementation.
Talos part is going to be changed pretty much.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: blockdevice/probe/probe.go [hunk 1]
```diff
@@ -34,106 +26,96 @@ type ProbedBlockDevice struct {
...
-
-	for _, info := range infos {
-		devpath := "/dev/" + info.Name()
+// SelectOption is a callback matcher for All block devices probes.
+type SelectOption func(device *ProbedBlockDevice) (bool, error)
 
-		var probed []*ProbedBlockDevice
-
-		probed, err = probeFilesystem(devpath)
+// WithPartitionsLabeled search for a block device which has partitions with some specific labels.
```
Comment (smira) [inline blockdevice/probe/probe.go:32]:
```suggestion
// WithPartitionsLabeled search for a block device which has partitions with some specific label.
```

File: blockdevice/probe/probe.go [hunk 2]
```diff
@@ -34,106 +26,96 @@ type ProbedBlockDevice struct {
...
+func WithPartitionsLabeled(label string) SelectOption {
```
Comment (smira) [inline blockdevice/probe/probe.go:33]:
```suggestion
func WithPartitionLabel(label string) SelectOption {
```

nit, I think this sounds better

File: blockdevice/probe/probe.go [hunk 3]
```diff
@@ -34,106 +26,96 @@ type ProbedBlockDevice struct {
...
-		}
+				if !add {
+					break
+				}
+			}
 
-		if sb.Is() {
-			return sb, nil
+			if add {
+				all = append(all, probed...)
```
Comment (smira) [inline blockdevice/probe/probe.go:71]: if several partitions match the options, is it going to add `probed` many times?

Does it add all of them despite only some of them matching?

File: blockdevice/probe/probe.go [hunk 4]
```diff
@@ -34,106 +26,96 @@ type ProbedBlockDevice struct {
...
+	if probed, err = All(WithPartitionsLabeled(value)); err != nil {
 		return nil, err
 	}
 
-	return filterByLabel(probed, value)
+	if len(probed) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	return probed[0], nil
```
Comment (smira) [inline blockdevice/probe/probe.go:92]: should we return an error if several partitions are found?

File: blockdevice/probe/probe.go (hunk 3 above)
Comment (Unix4ever) [reply blockdevice/probe/probe.go]: yeah, I've messed up here really bad. I've already fixed it locally, going to push it soon

File: blockdevice/probe/probe.go (hunk 4 above)
Comment (Unix4ever) [reply blockdevice/probe/probe.go]: yeah, let's do it

**Review by smira — APPROVED** [review]

**Pull request merged**